spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/service/xchange/TradeServiceXChangeImplementation.java

```diff
@@ -250,27 +254,37 @@ public final boolean cancelOrder(final String orderId) {
 
             // Query 1 week of trades.
             TradeHistoryParamsAll params = new TradeHistoryParamsAll();
-            Date startDate = DateUtils.addWeeks(new Date(), -1);
-            Date endDate = new Date();
+            Date now = TimeProvider.now();
+            Date startDate = DateUtils.addDays(now, -HOURS_IN_DAY);
             params.setStartTime(startDate);
-            params.setEndTime(endDate);
+            params.setEndTime(now);
+
+            Set<TradeDTO> results = new LinkedHashSet<>();
 
             // Set currency pairs (required for exchanges like Gemini or Binance).
             if (!currencyPairs.isEmpty()) {
-                params.setCurrencyPairs(currencyPairs.stream().map(currencyMapper::mapToCurrencyPair).collect(Collectors.toCollection(LinkedList::new)));
+
+                currencyPairs.forEach(pair -> {
+
+                    params.setCurrencyPair(currencyMapper.mapToCurrencyPair(pair));
+
+                    try {
+                        results.addAll(
+                                tradeService.getTradeHistory(params)
```

Comment:
If we have to make one call by currency pair, we will have to add getBucket().asScheduler().consume(1); to we won't be ban for calling too often the exchange